src/common/types/types.cpp

```diff
@@ -88,12 +105,38 @@ DataType Types::dataTypeFromString(const std::string& dataTypeString) {
         dataType.childType = std::make_unique<DataType>(
             dataTypeFromString(dataTypeString.substr(0, dataTypeString.size() - 2)));
         return dataType;
+    } else if (dataTypeString.ends_with(")")) {
+        dataType.typeID = TENSOR;
+        auto splitPos = dataTypeString.find(',');
+        auto childTypeStartPos = strlen("TENSOR") + 1; // The child type is followed by "TENSOR(".
+        dataType.childType = std::make_unique<DataType>(dataTypeFromString(
+            dataTypeString.substr(childTypeStartPos, splitPos - childTypeStartPos)));
+        dataType.numElementsInTensor = std::strtoll(
+            dataTypeString.substr(splitPos + 1, dataTypeString.size() - splitPos - 1).c_str(),
+            nullptr, 0 /* base */);
+        validateTensorType(dataType);
+        return dataType;
     } else {
         dataType.typeID = dataTypeIDFromString(dataTypeString);
     }
     return dataType;
 }
 
+void Types::validateTensorType(DataType& dataType) {
```

Comment:
I think this check should be moved to binder right before where you call dataTypeFromString

Reply:
moved to binder